File: Stage21/C/Problem24416.c
#include <stdio.h>

int countRec = 0;
int countDP = 0;

int fib(int n) {
    if (n == 1 || n == 2) {
        countRec++;
        return 1;
    }
    return fib(n - 1) + fib(n - 2);
}

int fibonacci(int n) {
    int f[50];
    f[1] = 1;
    f[2] = 1;

    for (int i = 3; i <= n; i++) {
        countDP++;
        f[i] = f[i - 1] + f[i - 2];
    }
    return f[n];
}

int main(void) {
    int n;
    scanf("%d", &n);

    fib(n);
    fibonacci(n);

    printf("%d %d\n", countRec, countDP);

    return 0;
}

File: Stage21/Python/Problem24416.py
import sys

def main():
    n = int(sys.stdin.readline().strip())

    fib = [0] * (n + 1)
    fib[1] = fib[2] = 1
    for i in range(3, n + 1):
        fib[i] = fib[i - 1] + fib[i - 2]

    count_rec = fib[n]
    count_dp = n - 2

    print(count_rec, count_dp)

if __name__ == "__main__":
    main()